internal/shim-sgx/src/handler/mod.rs

```diff
@@ -164,26 +210,39 @@ impl guest::Handler for Handler<'_> {
         fd: c_int,
         offset: off_t,
     ) -> sallyport::Result<NonNull<c_void>> {
-        Ok(NonNull::new(crate::heap::HEAP.write().mmap::<c_void>(
-            addr.map(|v| v.as_ptr() as usize).unwrap_or(0),
-            length,
-            prot,
-            flags,
-            fd,
-            offset,
-        )?)
-        .unwrap())
+        let addr = addr.map(|v| v.as_ptr() as usize).unwrap_or(0);
+
+        if addr != 0
+            || length == 0
+            || fd != -1
+            || offset != 0
+            || flags != MAP_PRIVATE | MAP_ANONYMOUS
+        {
+            return Err(ENOTSUP);
+        }
+
+        if prot & !(PROT_READ | PROT_WRITE | PROT_EXEC) != 0 {
+            return Err(ENOTSUP);
```

Comment:
Ok, I was wrong, this is really `EINVAL`

Reply:
fixed